### ISSUE TITLE
Keep raeper state inside Cassandra

### ISSUE THREAD

**Issue by nathan-gs** [spotify/cassandra-reaper#104]
Instead of relying on a outside data source, like Postgres, it would make sense to keep the state inside Cassandra, in a separate keyspace.

**Comment by Bj0rnen** [comment]
There would be 2 ways to go about this, when using one Reaper for many clusters:
* Configure a dedicated Reaper cluster, which holds this keyspace.
* Add all clusters to the Reaper's config, and let them all hold state of their own repairs.

The first one is pretty straight-forward. It's just another implementation of IStorage.

The second option is also clearly possible, considering there is no cross-cluster state. Except the info of which clusters there are, which would now be moved to the config. But it would not play well with IStorage, since Reaper wouldn't know how to map IDs to clusters. Either all clusters would have to be asked, or a bigger rewrite would be needed to accommodate for this.

Which one of these options did you have in mind?

**Comment by nathan-gs** [comment]
The second option would have my preference. 

Option 1 would also work in the mean time, especially if it can be the same cluster as it's monitoring.

**Comment by yingzong** [comment]
I disagree. It is a bad idea for any maintenance state to be stored in the system being maintained. It should be totally independent. DataStax's OpsCenter made that grievous mistake.
